File: core/src/main/java/fr/sncf/osrd/utils/graph/Graph.java
package fr.sncf.osrd.utils.graph;

import java.util.Collection;

/** This is the minimal interface we need to run pathfindings. */
public interface Graph<NodeT, EdgeT> {
    /** Returns the node placed at the end of the given edge */
    NodeT getEdgeEnd(EdgeT edge);

    /** Returns all the edges that start at the given node */
    Collection<EdgeT> getAdjacentEdges(NodeT node);
}

File: core/src/main/kotlin/fr/sncf/osrd/graph/Graph.kt
package fr.sncf.osrd.graph

/** This is the minimal interface we need to run pathfindings. */
interface Graph<NodeT, EdgeT, OffsetT> {
    /** Returns the node placed at the end of the given edge */
    fun getEdgeEnd(edge: EdgeT & Any): NodeT

    /** Returns all the edges that start at the given node */
    fun getAdjacentEdges(node: NodeT & Any): Collection<EdgeT>
}
